engine/src/snapshot_api/rocks_snapshot.rs

```diff
@@ -0,0 +1,382 @@
+use std::{
```

Comment:
Please add some tests to cover the logic in this file. Because memory_snapshot is too simple, only the rocks is required.

Reply:
Added